src/SymbolTable.ts

```diff
@@ -9,7 +9,8 @@ import { DynamicType } from './types/DynamicType';
  */
 export class SymbolTable {
     constructor(
-        parent?: SymbolTable | undefined
+        parent?: SymbolTable | undefined,
+        public identifier: string = ''
```

Comment:
Let's call this `name` instead, so it's less likely to be confused with the `Identifier` token types. Also, the type is implied by the default value, so no need for the explicit type.
```suggestion
        public name = ''
```

Reply:
changed